main.tf

```diff
@@ -41,18 +40,31 @@ resource "aws_route" "public_internet_gateway" {
   gateway_id             = "${aws_internet_gateway.this.id}"
 }
 
+resource "aws_vpn_gateway_route_propagation" "public" {
+  count = "${length(var.public_propagating_vgws)}"
+
+  route_table_id = "${aws_route_table.public.id}"
+  vpn_gateway_id = "${element(var.public_propagating_vgws, count.index)}"
+}
+
 #################
 # Private routes
 #################
 resource "aws_route_table" "private" {
   count = "${length(var.azs)}"
 
   vpc_id           = "${aws_vpc.this.id}"
-  propagating_vgws = ["${var.private_propagating_vgws}"]
 
   tags = "${merge(var.tags, var.private_route_table_tags, map("Name", format("%s-private-%s", var.name, element(var.azs, count.index))))}"
 }
 
+resource "aws_vpn_gateway_route_propagation" "private" {
+  count = "${length(var.private_propagating_vgws)}"
+
+  route_table_id = "${aws_route_table.private.id}"
```

Comment:
There are multiple private route tables, but one public, so I think this is more correct:
```
count = "${length(var.azs)}"
route_table_id = "${element(aws_route_table.private.*.id, count.index)}"
```

You can test it by setting `private_propagating_vgws = ["123", "456", "789"]`

Reply:
Ah, damn it, my fault.

But count on the AZs wouldn't fit too? IMHO every `private_propagating_vgws` entry should be attached to every private route table? What do you think?

Reply:
Ah, damn it, my fault.

But count on the AZs wouldn't fit too? This would propagate the first vgw to the first private subnet only and so on, or am I wrong? 

IMHO every `private_propagating_vgws` entry must be attached to every private route table. At least that's the old behaviour. Have to think about, how to do that